test/serial.setup.sh.bats

```diff
@@ -0,0 +1,150 @@
+load 'test_helper/common'
+
+# ! NEEDS SERIAL EXECTION - DO NOT RUN IN PARALLEL
```

Comment:
The diff for this file is misleading. Nothing changed except for this line.. but because of the renaming, git thinks this is a completely new file. Makes the diff appear large :D

Reply:
> but because of the renaming, git thinks this is a completely new file

This is going to be annoying, but I'd like to request you avoid this issue by separating out commits of renaming and then a commit for file change, or if it's easier at this point the other way around.

You can probably handle this without making it a pain to address. Try this:

1. Use a rename tool or script to remove the `serial.` / `parallel.` (or similar with the optional numbers) to revert the filename.
2. Commit just these file renames.
3. Commit a revert commit that does the opposite.
4. Try rebase that revert commit (original test names to new test names) so that it's the first commit in this PR. If this works you're done.
5. If it's not going to go smoothly, revert the existing commits (_in GitKraken I can just right-click a commit and create a revert commit, I assume there is a direct git command for this_). Then you can revert the revert :laughing: and a rebase that should be able to drop original commits and some of the reverts that follow... that should give a clean commit history now. This is fairly easy to do with GitKraken UI for interactive rebasing a range of commits, if you've got the commits pushed here I can handle that part if you like :+1: 

I remember for when we imported the Github Wiki git history to merge into our git repo history here (_via merge commit, which visually interleaved the commit history, but didn't rewrite the main project commit history_), we made sure to separate the file renames from content edits to ensure git could follow a file to it's original and diff properly :)

---

I won't have time to look at this until about 8 hours or so, but I'm curious about the current naming convention choice vs using folders for `serial` and `parallel`? Any benefit in prefixing filenames with it?

Perhaps you're using that from one of the BATS ENV or scripts on what to run? (_although from our end grouping by folder should be just as easy?_) I know that I'd like to eventually have some folders to group tests by topics/relevance.

I was planning for CI to have several runners/jobs that would run a subset of the tests (as a different form of parallelism), a good candidate would be the SSL tests, where one of those takes a huge amount of time to iterate through permutations of RSA + ECDSA and modern / intermediate (legacy) TLS levels.

Reply:
I wanted to create directories, but this caused the ´load` directive in BATS in conjunction with `test/test_helper/common.sh`. Maybe a future PR can take care of this, for now, we should be fine with a simple naming scheme that is somewhat similar to DNS.

---

I will take care  of the commit history.

Reply:
Re-wrote history and I hope it is now easier to review.

Reply:
> Re-wrote history and I hope it is now easier to review.

Hey @georglauterbach sorry for the late response. Looking over the commit history, I can see a few files in the commits were not just renamed but added changes in that same commit, which breaks the rename detection for git blame history.

It happened in a few commits, if I have your permission I don't mind correct this history and doing a force-push?

Reply:
Also to preserve this rename history, this PR will use a rebase merge, not a squash merge.

Reply:
> > Re-wrote history and I hope it is now easier to review.
> It happened in a few commits, if I have your permission I don't mind correct this history and doing a force-push?

Oh, I see. [`gitui`](https://github.com/extrawurst/gitui) showed them as renaming to me (maybe I did not pay enough attention). Please, go ahead :)

---

I will make sure resolve merge conflicts with `master` later.